### PR TITLE
Transpile

### PR DESCRIPTION
**Pull request by Howon**
run "opam install core"

### CODE REVIEW

**Comment by Howon** [comment]
@jibbenHillen @aethos do help please

File: src/oscar.ml [hunk 1]
```diff
@@ -45,24 +49,26 @@ let _ =
...
+         Semant.check ast; *)
+         match action with
+            Ast   -> ()
+          | Sast  -> print_endline (Sast.str_sprogram sprogram)
+          | _ ->
+              let program = Transpile.c_program sprogram in
+              let c_op = "-Wall -pedantic -fsanitize=address -std=c++1y -O2" in
+              let cxx_incls = "-I/usr/local/include/ -L/usr/local/lib/ " in
+              let cxx = sprintf "clang++ %s %s " c_op cxx_incls in
+              let ch = Unix.open_process_out cxx in
```
Comment (Howon) [inline src/oscar.ml]: @jibbenHillen  @aethos this part is broken can you fix it

File: src/analyzer.ml [hunk 2]
```diff
@@ -975,5 +935,8 @@ let check_program (p : program) (slib : program) =
     | 1 -> (List.rev smessages,
             List.rev sactors,
             List.rev (List.filter (fun sf -> not (sf.sv_name = "main")) sfunctions),
-            (List.find (fun sf -> sf.sv_name = "main") sfunctions).sv_init)
+            try
```
Comment (jibben) [inline src/analyzer.ml]: remove this logic

File: src/oscar.ml [hunk 3]
```diff
@@ -25,17 +28,19 @@ let _ =
...
-        (List.assoc Sys.argv.(1) [
-                ("-l", LLVM_IR);    (* Generate LLVM, don't check *)
-                ("-c", Compile);    (* Generate, check LLVM IR *)
-                ("-p", Ast);        (* Don't gen LLVM, just prettyprint ast *)
-                ("-s", Sast)        (* Don't generate LLVM, just prettyprint *)
-        ], open_in Sys.argv.(2))
+    else if arg_len > 2 then
+      try
+        (List.Assoc.find_exn [
+                ("-c", Compile);  (* Generate, check LLVM IR *)
```
Comment (jibben) [inline src/oscar.ml:33]: can we add a -l flag that will shit out llvm through clang

File: src/transpile.ml [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    String.concat "," (List.map (fun elem ->
+      "{ " ^ c_tup elem ^ " }"
+    ) tel)
+
+and c_texpr tex =
+  let (se, _) = tex in
+    (match se with
+        SInt_Lit _ | SDouble_Lit _ | SChar_Lit _
+      | SBool_Lit _ | SId _ | SNoexpr -> str_texpr tex
+      | SUnit_Lit _                -> "unit"
```
Comment (jibben) [inline src/transpile.ml]: "void"

File: src/transpile.ml [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+      | SString_Lit s              -> std ^ "string(\"" ^ s ^ "\")"
```
Comment (jibbenHillen) [inline src/transpile.ml]: This works properly with strings containing " \" "?

File: src/transpile.ml [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+and c_sstmts = function
+    SBlock sstmts   -> String.concat "\n" (List.map c_sstmt sstmts)
+  | _           -> ""
+
+and c_formal f = c_type (snd f) ^ " " ^ fst f
+
+and c_formals fs = String.concat "," (List.map c_formal fs)
+
+and c_lambda sfl =
```
Comment (jibben) [inline src/transpile.ml:94]: to fix: me

**Review by jibben — CHANGES_REQUESTED** [review]

File: src/oscar.ml (hunk 3 above)
Comment (aethos) [reply src/oscar.ml:33]: got it

Comment (jibben) [reply src/oscar.ml:33]: let's do this through a PR to master and just get this branch merged though

File: src/transpile.ml (hunk 6 above)
Comment (jibben) [reply src/transpile.ml:94]: again, to be fixed in PR

File: src/transpile.ml (hunk 5 above)
Comment (Howon) [inline src/transpile.ml]: wnna fix it?

**Review by jibben — APPROVED** [review]

**Pull request merged**